spec/feature/unauthenticated_guest_spec.rb

```diff
@@ -184,24 +182,94 @@
+                   address1: "124 Elm St",
+                   address2: nil,
+                   city: "Seattle",
+                   state: "WA",
+                   zip: 98106,
+                   shared_bathroom: true,
+                   user_id: 1)
     visit listings_path
-    find(:css, "#nightly_rate_15000[value='15000']").set(true)
-    click_link_or_button "Filter by price"
+    select('<$150', :from => 'nightly_rate')
+    click_link_or_button "Filter"
     expect(page).to have_content("Rainy day house")
     expect(page).not_to have_content("Super classy")
   end
 
+  it "can filter listings page by Nightly Rate, Max Guests,
+  Category and City at the same time" do
+    User.create(first_name: "John",
+                last_name: "Doe",
+                email: "joe@gmail.com",
+                password: "password",
+                about_me: "valid",
+                id: 1,
+                display_name: "joe")
+    Listing.create(title: "Rainy day room",
+                   description: "room in the rain",
+                   category_id: 2,
+                   max_guests: 2,
+                   nightly_rate: 20000,
+                   address1: "124 Elm St",
+                   address2: nil,
+                   city: "Seattle",
+                   state: "WA",
+                   zip: 98106,
+                   shared_bathroom: true,
+                   user_id: 1)
+    Listing.create(title: "Toga Party!",
+                   description: "Animal House",
+                   category_id: 1,
+                   max_guests: 2,
+                   nightly_rate: 20000,
+                   address1: "126 Elm St",
+                   address2: nil,
+                   city: "Denver",
+                   state: "CO",
+                   zip: 80222,
+                   shared_bathroom: true,
+                   user_id: 1)
+    Listing.create(title: "Sigma Alpha Epsilon",
+                   description: "Frat Room - Gotta be on the list bro.",
+                   category_id: 2,
+                   max_guests: 0,
+                   nightly_rate: 20000,
+                   address1: "125 Elm St",
+                   address2: nil,
+                   city: "Denver",
+                   state: "CO",
+                   zip: 80222,
+                   shared_bathroom: true,
+                   user_id: 1)
+    Listing.create(title: "Room with a view",
+                   description: "from space",
+                   category_id: 2,
+                   max_guests: 2,
+                   nightly_rate: 50000,
+                   address1: "125 Elm St",
+                   address2: nil,
+                   city: "Denver",
+                   state: "CO",
+                   zip: 80222,
+                   shared_bathroom: true,
+                   user_id: 1)
+    visit listings_path
+    fill_in('City', :with => 'Denver')
+    select('Room', :from => 'category_id')
+    select('1', :from => 'max_guests')
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.